src/wui/waresdisplay.cc

```diff
@@ -505,6 +508,72 @@ WaresDisplay::WaresDisplay(UI::Panel* const parent,
    : AbstractWaresDisplay(parent, x, y, tribe, type, selectable) {
 }
 
+StockMenuWaresDisplay::StockMenuWaresDisplay(UI::Panel* const parent,
+                                             const int32_t x,
+                                             const int32_t y,
+                                             const Widelands::Player& p,
+                                             const Widelands::WareWorker type)
+   : WaresDisplay(parent, x, y, p.tribe(), type, false), player_(p), solid_icon_backgrounds_(true) {
+}
+
+std::string StockMenuWaresDisplay::info_for_ware(const Widelands::DescriptionIndex di) {
+	const std::string text = WaresDisplay::info_for_ware(di);
+	if (solid_icon_backgrounds_) {
+		return text;
+	}
+
+	// Indicate trend over the last 5 minutes
+	const std::vector<uint32_t>& history = get_type() == Widelands::wwWARE ?
+	                                          *player_.get_ware_stock_statistics(di) :
+	                                          *player_.get_worker_stock_statistics(di);
+	const size_t nr_entries = history.size();
+
+	if (!nr_entries) {
+		// No records yet
+		return text;
+	}
+
+	const size_t kSampleEntriesForTrend = 5 * 60 * 1000 / Widelands::kStatisticsSampleTime.get();
+	const uint32_t last_amount =
+	   history[nr_entries < kSampleEntriesForTrend ? nr_entries - 1 :
+	                                                 nr_entries - kSampleEntriesForTrend];
+	const uint32_t current_amount = amount_of(di);
+
+	const UI::BuildingStatisticsStyleInfo& colors = g_style_manager->building_statistics_style();
+	const std::string indicator =
+	   current_amount < last_amount ?
+	      StyleManager::color_tag(_("↓"), colors.alternative_low_color()) :
+	      current_amount > last_amount ?
+	      StyleManager::color_tag(_("↑"), colors.alternative_high_color()) :
+	      StyleManager::color_tag(_("="), colors.alternative_medium_color());
```

Comment:
The icon width only allows four characters, so the indicator should be skipped for amount >= 1000. At this level the change would be unimportant anyway.